Simplenote/Classes/NSPredicate+Simplenote.swift

```diff
@@ -0,0 +1,62 @@
+import Foundation
+
+
+// MARK: - NSPredicate Methods
+//
+extension NSPredicate {
+
+    /// Returns a collection of NSPredicates that will match, as a compound, a given Search Text
+    ///
+    @objc(predicateForSearchText:)
+    static func predicateForSearchText(searchText: String) -> NSPredicate {
+        let words = searchText.trimmingCharacters(in: .whitespaces).components(separatedBy: .whitespaces)
+        var output = [NSPredicate]()
+
+        for word in words where !word.isEmpty {
+            let predicate = NSPredicate(format: "content CONTAINS[c] %@", word)
+            output.append(predicate)
+        }
```

Comment:
Not sure if this is the case but if you trim the text for `.whitespacesAndNewlines` then you can avoid the `where !word.isEmpty`. So you can reduce the code with `let output = words.map { NSPredicate(format: "content CONTAINS[c] %@", $0) }`. Have you tried this already?